File: fast_barcode_scanner/ios/Classes/types/PreviewConfiguration.swift
struct PreviewConfiguration {
    let width: Int32
    let height: Int32
    let targetRotation: Int
    let textureId: Int64

    var asDict: [String: Any] {
        ["width": height,
         "height": width,
         "targetRotation": targetRotation,
         "textureId": textureId,
         "analysisWidth": width,
         "analysisHeight": height]
    }
}

File: fast_barcode_scanner/android/src/main/kotlin/com/jhoogstraat/fast_barcode_scanner/types/PreviewConfiguration.kt
package com.jhoogstraat.fast_barcode_scanner.types

data class PreviewConfiguration(val textureId: Long, val targetRotation: Int, val width: Int,
                                val height: Int, val analysisWidth: Int,
                                val analysisHeight: Int) {
    fun toMap() = hashMapOf(
            "textureId" to textureId,
            "targetRotation" to targetRotation,
            "width" to width,
            "height" to height,
            "analysis" to "${analysisWidth}x${analysisHeight}",
            "analysisWidth" to analysisWidth,
            "analysisHeight" to analysisHeight)
}
